Paa1006-Assignment4/src/paa1006/assignment4/Paa1006Assignment4.java

```diff
@@ -0,0 +1,144 @@
+                Scanner keyboard2 = new Scanner(System.in);
+                String word;
+                System.out.print("Enter a string: ");
+                word = keyboard2.next();
+                
+                // Call method to get value of string
+               ValueOfString(word);
+               break;
+            case 2:
+                // Call method to Display Date and time
+                DisplayDateTime();
+                break;
+            case 3:
+                // Call method to display name
+                DisplayName();
+                break;
+            case 4:
+                // Call method to display IDE
+                DisplayIDE();
+                break;
+            case 5:
+                // Call method to display Github user name
+                DisplayGithub();
+                break;
+            case 6:
+                // Call method to display close program message
+                exitP = CloseProgram();
+                break;
+            default:
+                // Call method to display wrong input
+                WrongInput();
+        }
+    }while(exitP == false);
+}
+    
+    /**
+     * Method which takes a string, and finds its integer value
+     * @param word String
+     */
+    public static void ValueOfString(String word){
+        // Initialize a counter for value of string
+        int valueTotal = 0;
+        
+        // Loop to add up chars in the string
+        for(int i = 0;i < word.length();i++){
+          valueTotal += word.charAt(i);
+        }
+       
+         // Print out information
+        System.out.println("The word is " + word);
+        for(int i = 0;i < word.length();i++){
+         System.out.println(word.charAt(i) + " = " + (int)word.charAt(i));
+        }            
+        System.out.println("Total = " + valueTotal);      
+    }
+    
+    /**
+     * Display Date and Time
+     */
+    public static void DisplayDateTime(){
+        DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+        Date date = new Date();
+        System.out.println("The current date and time is: " + dateFormat.format(date));
+    }
+    
+    /**
+     * Display name
+     */
+    public static void DisplayName(){
+        System.out.println("The programmer's name is Patrick Antoniak.");
+    }
+    
+    /**
+     * Display IDE
+     */
+    public static void DisplayIDE(){
+        System.out.println("The IDE used to program this was Netbeans.");
+    }
+    
+    /**
+     * Display Github user name
+     */
+    public static void DisplayGithub(){
+        System.out.println("The programmer's Github username is Paa1006.");
+    }
+    
+    /**
+     * Close the program by returning the value to end the loop
+     * @return Boolean True
+     */
+    public static Boolean CloseProgram(){
+        Boolean exit;
+        
+        System.out.println("The program is now closing...");
+        exit = true;
+        return exit;
```

Comment:
You could also just return the `true` literal. Functionally it is the same but by number of operations.
Below is mostly assembler aside from `malloc` which is a C function but I just want to illustrate the steps the computer is taking.

1. `malloc` (memory allocation) for a boolean (1 bit)
1. `mov` (assign value to memory) of the literal `true`
1. `mov` (read value from memory)
1. `ret` (return) with reference to memory

Just using the literal in the return.

1. `ret` return with value of cached literal `true`

That was a big explanation but its nice to think about what the computer could be doing.

Here is the cool part. It is likely the compiler will see this an go:
- Within the scope of CloseProgram the value was declared initialized and there are no branches (loop or conditions) where it was modified. Also the value is a cached literal. Thus it may just optimize the code by returning the literal directly. Since we will never likely know all the patterns the compiler will use for optimization its good to try and think for it.